lib/provided_promise.js

```diff
@@ -0,0 +1,18 @@
+'use strict';
+
+const assert = require('assert');
+
+class ProvidedPromise {
```

Comment:
to reiterate my point above, this class is not the actual provided Promise, but rather something a user can provide a Promise to/with. I think a name like `PromiseProvider` is more appropriate.

Reply:
no preference here 👍